### PR TITLE
AK+LibWeb: Port {HTML,UIEvents,XHR}::EventNames to new FlyString

### PR DESCRIPTION
**Pull request by kennethmyhra**

### CODE REVIEW

**Comment by trflynn89** [comment]
This isn't quite right - these were previously fly strings, thus allocated just once. Now we allocate these names every time we use them. Instead, these should be changed from `DeprecatedFlyString` to `FlyString`.

**Comment by kennethmyhra** [comment]
Thanks, I'll continue down that path then. Setting this to draft in the meanwhile

**Comment by kennethmyhra** [comment]
Now correctly porting {HTML,UIEvents,XHR}::EventNames to FlyString.

File: Userland/Libraries/LibWeb/HTML/EventNames.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ ErrorOr<void> initialize_strings()
     VERIFY(!s_initialized);
 
 #define __ENUMERATE_HTML_EVENT(name) \
-    name = #name;
+    name = TRY(String::formatted(#name));
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/EventNames.cpp:21]: ...formatting a single string literal?

File: Userland/Libraries/LibWeb/UIEvents/EventNames.cpp [hunk 2]
```diff
@@ -18,7 +18,7 @@ ErrorOr<void> initialize_strings()
     VERIFY(!s_initialized);
 
 #define __ENUMERATE_UI_EVENT(name) \
-    name = #name;
+    name = TRY(String::formatted(#name));
```
Comment (linusg) [inline Userland/Libraries/LibWeb/UIEvents/EventNames.cpp:21]: Equally :thonk:

File: Userland/Libraries/LibWeb/XHR/EventNames.cpp [hunk 3]
```diff
@@ -18,7 +18,7 @@ ErrorOr<void> initialize_strings()
     VERIFY(!s_initialized);
 
 #define __ENUMERATE_XHR_EVENT(name) \
-    name = #name;
+    name = TRY(String::formatted(#name));
```
Comment (linusg) [inline Userland/Libraries/LibWeb/XHR/EventNames.cpp:21]: And here

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Libraries/LibWeb/HTML/EventNames.cpp (hunk 1 above)
Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/HTML/EventNames.cpp]: You are absolutely right, I had gone through this so many times to make it work that I had almost given up on it. Hopefully the current change is better.

```cpp
#define __ENUMERATE_HTML_EVENT(name) \
    name = TRY(String::from_utf8(StringView { #name, strlen(#name) }));
    ENUMERATE_HTML_EVENTS
#undef __ENUMERATE_HTML_EVENT
```

Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/HTML/EventNames.cpp]: Updated it to be `FlyString::from_utf8()`

File: Userland/Libraries/LibWeb/HTML/EventNames.cpp [hunk 4]
```diff
@@ -18,7 +18,7 @@ ErrorOr<void> initialize_strings()
     VERIFY(!s_initialized);
 
 #define __ENUMERATE_HTML_EVENT(name) \
-    name = #name;
+    name = TRY(FlyString::from_utf8(StringView { #name, strlen(#name) }));
```
Comment (linusg) [inline Userland/Libraries/LibWeb/HTML/EventNames.cpp:21]: Still seems overly complicated - just `from_utf8(#name##sv)` is fine, no?

Comment (trflynn89) [reply Userland/Libraries/LibWeb/HTML/EventNames.cpp:21]: Can also do `TRY(#name##_fly_string)` as we did here:

https://github.com/SerenityOS/serenity/blob/8a8340b3cdda0ab0dfd1dad734db933400422cfa/Userland/Libraries/LibWeb/DOM/MutationType.cpp#L20

Comment (kennethmyhra) [reply Userland/Libraries/LibWeb/HTML/EventNames.cpp]: That's much nicer. Updated

**Review by linusg — APPROVED** [review]
that's very nice progress!

**Pull request merged**